### ISSUE TITLE
As a content editor, I want to embed external content in editorial and other pages, so that I can include dynamic content in essays.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#731]
## testing notes
When you edit a page in wagtail and add a new block, you should now have the option to choose "embed".
Test editing a page to add an embed block, and add the url you want to embed; you can try testing with any of these:
  - https://princeton-cdh.github.io/literary-right-bank/
  - https://ppa-collections-treemap2.glitch.me/
  - https://ppa-collection-stripes4.glitch.me
  
Test both preview and publish to see the embedded content and check how it renders. Also check regular content pages and editorial pages.

Note: these embeds may not work if you try to add multiple of them to the same page due to differences in javascript dependencies, but if so, that's specific to the embedded content and not a problem for this feature. 

Also note: I haven't enabled any other embed finders at the moment, because I don't know that any of them are relevant for us. But here is the full list of embed finders provided by wagtail, in case you want to look. We can always expand the list later if we need one of these.

**Comment by jkotin** [comment]
Working on the test site, I'm running into some issues.  I see the embed option, and add the url for the right bank map. I save to draft, no problems so far. But when I try to preview after saving as draft, I get a "Bad Request (400)" message. If I publish, the resulting page doesn't include the map, just a blank space where the map should be:

https://test-shakespeareandco.cdh.princeton.edu/analysis/2021/03/literary-left-bank/

I'm sure I'm just missing something.  To embed the map, do I just add the link:

https://princeton-cdh.github.io/literary-right-bank/

**Comment by rlskoeser** [comment]
@jkotin that should work, but I see that there are some content protection errors. I don't know if something changed, or the config is different in qa than in my dev. I can investigate.

**Comment by rlskoeser** [comment]
@jkotin Nick helped me understand the workaround suggested in the d3 documentation and I've updated Moacir's code to avoid the CSP problem, which Moacir has reviewed and merged in. Do you want to confirm it's working now on the test site?

**Comment by jkotin** [comment]
Everything looks good when I publish the essay on the test site.  The "preview" function in Wagtail, however, isn't working.  I get a "Bad Request (400)" error.  The preview function isn't essential, however; I am happy to close.

**Comment by jkotin** [comment]
The PPA embeds don't seem to appear on the test site when published.
<img width="1119" alt="Screen Shot 2021-03-31 at 11 25 19 PM" src="https://user-images.githubusercontent.com/12616674/113239291-7618ce00-9278-11eb-830a-d2164b8f75d3.png">

**Comment by rlskoeser** [comment]
The 400 bad request worried me, but I figured it out — it was a site name misconfiguration in wagtail (data had been copied from production and not updated for qa), and django is configured (for security reasons) only to respond to certain hostnames. I corrected the wagtail site url and the 400 error is resolved now.

And, of course, the PPA embeds don't work because they have different javascript requirements which I didn't add to the S&co CSP list. Sorry for asking you to test with them — shall we ignore that error and close this?
